crates/storage/codecs/derive/src/lib.rs

```diff
@@ -51,15 +57,49 @@ mod compact;
 ///   efficient decoding.
 #[proc_macro_derive(Compact, attributes(maybe_zero, reth_codecs))]
 pub fn derive(input: TokenStream) -> TokenStream {
-    let is_zstd = false;
-    compact::derive(input, is_zstd)
+    compact::derive(input, None)
 }
 
 /// Adds `zstd` compression to derived [`Compact`].
-#[proc_macro_derive(CompactZstd, attributes(maybe_zero, reth_codecs))]
+#[proc_macro_derive(CompactZstd, attributes(maybe_zero, reth_codecs, reth_zstd))]
 pub fn derive_zstd(input: TokenStream) -> TokenStream {
-    let is_zstd = true;
-    compact::derive(input, is_zstd)
+    let derive_input = {
+        let input = input.clone();
```

Comment:
wonder if we need to do this here or if we can do this 

https://github.com/paradigmxyz/reth/blob/73f1583455ace8e9012271e40bd4fe1b42d87715/crates/storage/codecs/derive/src/compact/mod.rs#L46-L46

and get rid of the clone?
unsure though, we don't use zsdt derive a lot so this seems fine anyway

Reply:
Updated `compact::derive` to just accept a pre-parsed `DeriveInput`, we don't need clone that way